base/test_main_docker.go

```diff
@@ -0,0 +1,9 @@
+// +build cb_sg_dockertest
+
+package base
+
+import "testing"
+
+func InitTestMain(t *testing.M) {
```

Comment:
Not sure there's much value in the intermediate `func InitTestMain(t *testing.M) {`

How about guarding the existing `TestMain` with the build tag?

Reply:
I felt like the intermediate one would be better. Rather than adding the build tag and check in every package which would be required to do it directly on the TestMain right?

Reply:
I would opt for including the `// +build cb_sg_dockertest` build tag in every package and call `NewDockerTest(m)`. You won't need a corresponding `!cb_sg_dockertest` file anywhere, as the default does what you want it to.

Reduces the number of files required, and reduces the number of indirection.